### PR TITLE
Fix file decode

### PR DESCRIPTION
**Pull request by quaternion**
Fix `ArgumentError: invalid byte sequence in UTF-8` when decoding real files in ruby-1.9.2p312:

``` ruby
File.bdecode('bencode.txt')
# or
file = File.open('bencode.txt', 'rb')
file.bdecode
```

Add necessary tests.
Remove `force_encoding('UTF-8')` from `File#bdecode`.
Fix minor error in readme file.
Also had update gem dependencies.

### CODE REVIEW

File: spec/bencodr/io_spec.rb [hunk 1]
```diff
@@ -1,35 +1,62 @@
...
-    File.bencode(@file, @object)
+  let(:file)   { Tempfile.new('test.bencodr') }
+  let(:object) { "string" }
+
+  describe ".bencode" do
+    it "should encode object to file" do
+      File.bencode(file, object)
+      file.rewind
+      file.read.should == "6:string"
+    end
```
Comment (blatyo) [inline spec/bencodr/io_spec.rb:18]: Cool trick. Didn't know about `Tempfile`.

**Comment by blatyo** [comment]
This removes the only dependency I had on any ruby 1.9 features which is cool. I'll have to get this project setup with travis, to make sure this doesn't break for older rubies.

**Comment by quaternion** [comment]
+1 setup with travis.

**Comment by blatyo** [comment]
http://travis-ci.org/#!/blatyo/bencodr

There are some 1.8 things that are preventing me from pushing the gem. One is:

```
Failure/Error: File.bencode(file, object)
63     TypeError:
64       can't convert Tempfile into String
65     # ./lib/bencodr/io.rb:7:in `initialize'
66     # ./lib/bencodr/io.rb:7:in `open'
67     # ./lib/bencodr/io.rb:7:in `bencode'
68     # ./spec/bencodr/io_spec.rb:15
```

Most of the others are just odd differences between rubies where I'll probably have specs for different versions of ruby.

**Comment by quaternion** [comment]
Good, 1.9.2 and 1.9.3 work correctly. I'll try to switch to 1.8.7 and find out the cause of errors.

**Pull request merged**